server/src/main/java/org/uiautomation/ios/utils/SimulatorSettings.java

```diff
@@ -194,13 +198,28 @@ private String getSDKSubfolder(String desiredSDKVersion) {
     return SDK_LOCATION + "iPhoneSimulator" + suffix + ".sdk";
   }
 
-  private String determineDeviceUUID() {
+  private String determineDeviceUUID() throws NotActiveException {
     DeviceUUIDsMap uuidsMap = new DeviceUUIDsMap();
     uuidsMap.loadData();
     String uuid = uuidsMap.getUUID(exactSdkVersion, deviceName);
     if (uuid == null) {
-      System.err.println("Unable to get UUID for device " + deviceName + " with SDK " + exactSdkVersion);
+      throw new NotFoundException("Couldn't find UUID for device " + deviceName + " with SDK " + exactSdkVersion);
```

Comment:
`NotFoundException` is of type `RuntimeException` which means your change is taking a message which was previously printed to the screen and may now go unnoticed. Should we `log.info` or `log.warn` too?